atlas-spec/mpc-engine/src/primitives/auth_share.rs

```diff
@@ -1,13 +1,32 @@
 //! This module defines the interface for share authentication.
 use super::mac::{Mac, MacKey};
 
-/// An authenticated share of a bit.
-#[allow(dead_code)] // TODO: Remove this later.
-pub struct AuthShare {
-    /// Party i's share of the bit
-    pub(crate) share: bool,
-    /// MACs on the shared bit provided by the other parties
+/// A bit held by a party with a given ID.
+#[derive(Debug, Clone)]
+pub struct Bit {
+    pub(crate) id: BitID,
+    pub(crate) value: bool,
+}
+#[derive(Debug, Clone)]
+/// A bit identifier.
+///
+/// This is unique per party, not globally, so if referring bits held by another
+/// party, their party ID is also required to disambiguate.
+pub struct BitID(pub(crate) usize);
```

Comment:
Is there a reason for usize and not a fixed size int?

Reply:
At the moment the `BitID` isn't used, but as soon as we move to function-dependent preprocessing it will be used to index into the list of gates in the circuit, so I made it a `usize`.